tests/integration/aclgen_test.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2015 The Capirca Project Authors All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import sys
+import logging
+import os
+import inspect
+import shutil
+import tempfile
+from cStringIO import StringIO
+import filecmp
+
+import aclgen
+from lib import policy
+from lib import naming
+
+
+class Test_AclGen_ensure_demo_files_work(unittest.TestCase):
+  """Ensure Capirca demo runs successfully out-of-the-box."""
+
+  def setUp(self):
+    self.iobuff = StringIO()
+    logger = logging.getLogger()
+    logger.level = logging.DEBUG
+    self.s = logging.StreamHandler(self.iobuff)
+    self.s.level = logging.DEBUG
+    logger.addHandler(self.s)
+
+    # Capirca only writes if files have changed, so write out to a new
+    # temp dir for each test run.
+    self.output_dir = tempfile.mkdtemp()
+
+    curr_dir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+    self.root_dir = os.path.realpath(os.path.join(curr_dir, '..', '..'))
+    self.policies_dir = os.path.join(self.root_dir, 'policies')
+    self.defs_dir = os.path.join(self.root_dir, 'def')
+
+  def tearDown(self):
+    # print self.iobuff.getvalue().split("\n")
+    # Remove the directory after the test
+    shutil.rmtree(self.output_dir)
+
+  def test_smoke_test_generates_successfully(self):
+    args = [
+      'program',  # Dummy value for gflags, which expects the program name to be the first entry.
+      '--base_directory={0}'.format(self.policies_dir),
```

Comment:
using .format here is a bit overkill. String format is slower than a simple modulo format. We can leave this how it is and save it for a cleanup afterwards.